### DISCUSSION TITLE
RedirectResponse would not work!

### DISCUSSION THREAD

**Discussion by dr-sandipan-chowdhury** [zauberzeug/nicegui#3243] [Q&A]
### Question

```py
from nicegui import ui, app 
from starlette.requests import Request 
from starlette.responses import RedirectResponse, JSONResponse

@app.route('/login', methods=['POST'])
async def login(req: Request):
  print(req.method)
  form = await req.form()
  x = form.get('credential')
  print(x)
  return RedirectResponse('/welcome')

@ui.page('/welcome')
def wlc():
  ui.label('Welcome User')
```
![Screenshot from 2024-06-18 23-11-30](https://github.com/zauberzeug/nicegui/assets/75874218/56e02748-bc09-41b7-8102-4ca0dd968e34)

**Comment by rodja** [comment]
Sorry I can not reproduce your issue. When I open the /welcome url, I see the label. And the /login shows the error json. Which is expected because you only specified the POST method for this route. The browser does a GET request which is then producing the error json.

**Comment by dr-sandipan-chowdhury** [comment]
```py
from nicegui import ui, app 
from starlette.requests import Request 
from starlette.responses import RedirectResponse, JSONResponse

redirect_template = '<!DOCTYPE html><html><head><meta http-equiv="refresh" content="0; url={url}"></head><body></body></html>'

@app.route('/login', methods=['POST'])
async def login(req: Request):
  print(req.method)
  form = await req.form()
  x = form.get('credential')
  print(x)
  return Response(redirect_template.format(url='/welcome'))

@ui.page('/welcome')
def wlc():
  ui.label('Welcome User')
```
What I was trying to do is: Google One Tap sign-in integration. For that they need you to provide an endpoint that is capable of taking POST method. There they give you the user's credential and email. As, ui.page does not support post request, I used fastapi app.post roughly... because I know very little about fastapi... But I could not go to any ui.page from the app's context. So, I did this work-around where I sent browser to redirect to the desired url.

**Comment by dr-sandipan-chowdhury** [comment]
Is there any other alternative(easy) so that I can get to '/welcome' page from the `app.post`

**Comment by rodja** [comment]
Ah, now I understand whats going on. Google's One-Tap is taking the response of the POST request and displays it as content to the user. Redirect responses seems not to be supported by Google. Thats why your `redirect_template` works and is a okay-isch solution. I just started https://github.com/zauberzeug/nicegui/pull/3244 to create a One-Tap Auth example. There I use some client/page magic to create Welcome NiceGUI page content directly in the POST request. What do you think?
